### PR TITLE
Add send mail support, add request attributes to saga transaction con…

### PR DESCRIPTION
**Pull request by igordumchykov**
- add mail support was moved from master. Reason: commons mail is available starting from xm commons 2.1.0, therefore there are issues with starting up microservice on jre8

- async lep execution does not allow retrieve http request (stored in ThreadLocal) from lep

- xm commons was downgraded to 2.1.2 because 2.1.3 could not be find (before merge need upgrade again)

- PrivilegeDescriptor available from 2.1.3, therefore was removed from resource services (before merge will be restored once 2.1.3 will be available)

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java [hunk 1]
```diff
@@ -33,15 +32,13 @@
     @Timed
     @PreAuthorize("hasPermission({'sagaTransaction': #sagaTransaction}, 'ACTIVATION.TRANSACTION.CREATE')")
     @PostMapping("/transaction")
-    @PrivilegeDescription("Privilege to create a new saga transaction")
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java]: what reason to remove description?

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java [hunk 2]
```diff
@@ -33,15 +32,13 @@
...
     public ResponseEntity<SagaTransaction> createSagaTransaction(@RequestBody SagaTransaction sagaTransaction) {
         return ResponseEntity.ok(sagaService.createNewSaga(sagaTransaction));
     }
 
     @Timed
     @PreAuthorize("hasPermission({'id': #id}, 'ACTIVATION.TRANSACTION.CONTINUE')")
     @PostMapping("/task/{id}/continue")
-    @PrivilegeDescription("Privilege to continue suspended task")
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java]: what reason to remove description?

File: src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java [hunk 3]
```diff
@@ -51,15 +48,13 @@
     @Timed
     @PostMapping("/transaction/{id}/cancel")
     @PreAuthorize("hasPermission({'id': #id}, 'ACTIVATION.TRANSACTION.CANCEL')")
-    @PrivilegeDescription("Privilege to cancel saga transaction")
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/activation/web/rest/SagaTransactionResource.java]: what reason to remove description?

**Comment by igordumchykov** [comment]
> what reason to remove privilege description?

I didn`t find it in xm commons 2.1.3 and 2.1.2. I mentioned in the last comment to my PR that I will add it before PR will be merged.

File: gradle.properties [hunk 4]
```diff
@@ -32,7 +32,7 @@ spring_no_http_plugin_version=0.0.3.RELEASE
 checkstyle_version=8.24
 openapi_plugin_version=4.2.2
 
-xm_commons_version=2.1.3
+xm_commons_version=2.1.2
```
Comment (sergeysenja1992) [inline gradle.properties]: https://mvnrepository.com/artifact/com.icthh.xm.commons/xm-commons-permission/2.1.3
2.1.3 already in maven )

Comment (igordumchykov) [reply gradle.properties]: Done

**Comment by igordumchykov** [comment]
Outdated, part of #54

**Pull request closed without merging**